src/test/java/org/mockito/internal/configuration/plugins/PluginLoaderTest.java

```diff
@@ -0,0 +1,87 @@
+package org.mockito.internal.configuration.plugins;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.io.IOUtil;
+import org.mockitoutil.ClassLoaders;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test PluginLoader. This test check that loaded plugin can be assigned to fields of classes what loaded by
+ * application class loader
+ */
+public class PluginLoaderTest {
+
+    public static final String PLUGIN_PATH = "mockito-extensions/org.mockito.internal.configuration.plugins.PluginInterface";
+    private PluginLoader pluginLoader;
+
+    @Before
+    public void setUp() {
+        pluginLoader = new PluginLoader(new DefaultPluginSwitch());
+    }
+
+    @Test
+    public void should_load_custom_plugin_without_context_classloader() {
+
+        Object pluginInterface = loadPlugin();
+
+        assertThatLoadedPluginAssignable(pluginInterface);
+    }
+
+
+    @Test
+    public void should_load_custom_plugin_with_custom_context_classloader() throws Exception {
+
+        ClassLoader classLoader = ClassLoaders.inMemoryClassLoader().withParent(getParentClassLoader())
+                                          .withClassDefinition("", readResource()).build();
```

Comment:
The `.withClassDefinition("", readResource())` doesn't register the class. It simple put value of `
readResource()` to map with key empty string as key. 
`private Map<String , byte[]> inMemoryClassObjects = new HashMap<String , byte[]>();`

And it's used only in to cases: 

-  find class with name, but empty string is invalid class name value, so it should never be called and if it called then `Exception`is correct behaviour for this case 
- getResources - then `Enumeration` is return. 

I could use not empty string, but `"mockito-extensions/org.mockito.internal.configuration.plugins.PluginInterface"`, but meaning will be same.